feat: add Events Calendar

conventoangelo opened the pull request:
This pull request introduces a new `Calendar` component to the `website-frontend` project and integrates it into the events page.

* Added a new `Calendar` component to display events in a calendar view. This includes functions for navigating between months, formatting event dates and times, and rendering events with their details.
* Replaced the `LoadMore` button and the list of events with the new `Calendar` component to enhance the user experience when browsing events.

eyronjuude commented:
@conventoangelo I really like your implementation for the Events Calendar! I'd like to propose some changes, though if it's at your convenience to do so in this PR:

1. Given the Upcoming, Past, and All filters, is it possible to limit the selectable months based on those filters? For example, for Upcoming, given that today's March 8, 2025, it's only possible to select March 2025 onwards in the Months selector. For Past, still March, but from March 202,5 there is no option to go past March 2025. Then, the default functionality for the Months selector when All is selected.
  - TL;DR: Upcoming - Current Month + Current Year onwards. Past - Current Month + Current Year backwards. All - default calendar functionality
2. I'd prefer to use `line-clamp` for Events Content so that events cards will not take too much space in the calendar — either `line-clamp-2` or `line-clamp-3` works based on the project specifications.
3. For events tags display, use `event_tags.name` instead of accessing obsolete `tags` fields.

In addition, I am not exactly sure what TBD does in the context of the timeframe for an event, specifically in this example:

![image](https://github.com/user-attachments/assets/71fe3778-83e9-4dde-961b-2c6d0c37792b)

Overall, I like the calendar design. It fits pretty well with the client's description of the calendar.

conventoangelo commented:
@eyronjuude thanks for reviewing the PR. To address your issues/suggestions:

1. I personally think the upcoming, past, and all filters should be deprecated or removed, as they don't offer additional functionality now that the calendar has been implemented. They would likely introduce more confusion going forward.
2. The events content already uses line clamp. There might have been an error in its implementation. I've set it to `line-clamp-1` if the event has a location and `line-clamp-2` if it doesn't. The previous logic wasn't working properly, but this has been fixed.
3. I have already updated the calendar to use `event_tags.name`.
4. TBD stands for "To Be Determined." It appears when an event does not specify an `end_date`. I've added a note in Directus explaining how to create an all-day event. The TBD indicator is particularly useful since we still need to display the start time of the event rather than simply defaulting to "all-day."

eyronjuude commented:
> @eyronjuude thanks for reviewing the PR. To address your issues/suggestions:
> 
> 1. I personally think the upcoming, past, and all filters should be deprecated or removed, as they don't offer additional functionality now that the calendar has been implemented. They would likely introduce more confusion going forward.
> 2. The events content already uses line clamp. There might have been an error in its implementation. I've set it to `line-clamp-1` if the event has a location and `line-clamp-2` if it doesn't. The previous logic wasn't working properly, but this has been fixed.
> 3. I have already updated the calendar to use `event_tags.name`.
> 4. TBD stands for "To Be Determined." It appears when an event does not specify an `end_date`. I've added a note in Directus explaining how to create an all-day event. The TBD indicator is particularly useful since we still need to display the start time of the event rather than simply defaulting to "all-day."

Thanks for the insights! Btw @VeeIsForVictor, I agree with the removal of upcoming, past, and all filters, but we may need your go signal on this as we might deviate a little bit from the proposed UAT user stories.

For now, I'll make an approved mark on this one. Good work!

eyronjuude reviewed (approved):
> @eyronjuude thanks for reviewing the PR. To address your issues/suggestions:
> 
> 1. I personally think the upcoming, past, and all filters should be deprecated or removed, as they don't offer additional functionality now that the calendar has been implemented. They would likely introduce more confusion going forward.
> 2. The events content already uses line clamp. There might have been an error in its implementation. I've set it to `line-clamp-1` if the event has a location and `line-clamp-2` if it doesn't. The previous logic wasn't working properly, but this has been fixed.
> 3. I have already updated the calendar to use `event_tags.name`.
> 4. TBD stands for "To Be Determined." It appears when an event does not specify an `end_date`. I've added a note in Directus explaining how to create an all-day event. The TBD indicator is particularly useful since we still need to display the start time of the event rather than simply defaulting to "all-day."

Thanks for the insights! Btw @VeeIsForVictor, I agree with the removal of upcoming, past, and all filters, but we may need your go signal on this as we might deviate a little bit from the proposed UAT user stories.

For now, I'll make an approved mark on this one. Good work!

VeeIsForVictor reviewed (approved):
I'll issue my approval on this, but I agree with the point on removing the previous event filter system. Let's handle that in a fix request though, so we can merge this in ASAP. LGTM! :rocket: :rocket: :rocket:
